File: lib/views/herd/animal_detail_view.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:go_router/go_router.dart';
import '../../utils/theme.dart';
import '../../widgets/common_widgets.dart';
import '../../providers/providers.dart';
import '../../models/database.dart';

class AnimalDetailView extends ConsumerWidget {
  final String animalId;

  const AnimalDetailView({super.key, required this.animalId});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final animalAsync = ref.watch(animalByIdProvider(animalId));
    final treatmentsAsync = ref.watch(treatmentRecordsProvider(animalId));
    final calvingsAsync = ref.watch(calvingRecordsProvider(animalId));
    final weightsAsync = ref.watch(weightRecordsProvider(animalId));

    return animalAsync.when(
      data: (animal) {
        if (animal == null) {
          return Scaffold(
            appBar: AppBar(title: const Text('Animal Not Found')),
            body: const Center(child: Text('Animal not found')),
          );
        }

        return Scaffold(
          appBar: AppBar(
            title: const Text('Animal Details'),
            actions: [
              TextButton(
                onPressed: () => context.push('/animal/$animalId/edit'),
                child: const Text('Edit'),
              ),
            ],
          ),
          body: SingleChildScrollView(
            padding: const EdgeInsets.all(AppTheme.standardPadding),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                _buildHeader(animal),
                const SizedBox(height: 16),
                _buildInfoSection(animal),
                const SizedBox(height: 16),
                if (animal.sex == AnimalSex.cow)
                  calvingsAsync.when(
                    data: (calvings) => _buildCalvingSection(context, calvings),
                    loading: () =>
                        const LoadingCard(title: 'Loading calvings...'),
                    error: (e, _) => ErrorCard(message: e.toString()),
                  ),
                const SizedBox(height: 16),
                treatmentsAsync.when(
                  data: (treatments) =>
                      _buildTreatmentSection(context, treatments),
                  loading: () =>
                      const LoadingCard(title: 'Loading treatments...'),
                  error: (e, _) => ErrorCard(message: e.toString()),
                ),
                const SizedBox(height: 16),
                weightsAsync.when(
                  data: (weights) => _buildWeightSection(context, weights),
                  loading: () => const LoadingCard(title: 'Loading weights...'),
                  error: (e, _) => ErrorCard(message: e.toString()),
                ),
                if (animal.notes.isNotEmpty) ...[
                  const SizedBox(height: 16),
                  _buildNotesSection(animal.notes),
                ],
                const SizedBox(height: 24),
                _buildDeleteButton(context, ref, animal),
              ],
            ),
          ),
        );
      },
      loading: () => Scaffold(
        appBar: AppBar(),
        body: const Center(child: CircularProgressIndicator()),
      ),
      error: (e, _) => Scaffold(
        appBar: AppBar(),
        body: Center(child: Text('Error: $e')),
      ),
    );
  }

  Widget _buildHeader(Animal animal) {
    return AppCard(
      child: Column(
        children: [
          Row(
            children: [
              Container(
                width: 80,
                height: 80,
                decoration: BoxDecoration(
                  color: AppTheme.earthGreen.withValues(alpha: 0.1),
                  borderRadius: BorderRadius.circular(16),
                ),
                child: Icon(
                  _getSexIcon(animal.sex),
                  size: 44,
                  color: AppTheme.earthGreen,
                ),
              ),
              const SizedBox(width: 16),
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      animal.tagNumber,
                      style: const TextStyle(
                        fontSize: 24,
                        fontWeight: FontWeight.bold,
                      ),
                    ),
                    Row(
                      children: [
                        Text(
                          animal.sex.label,
                          style: TextStyle(color: Colors.grey[600]),
                        ),
                        if (animal.breed.isNotEmpty) ...[
                          Text(' • ', style: TextStyle(color: Colors.grey[600])),
                          Text(
                            animal.breed,
                            style: TextStyle(color: Colors.grey[600]),
                          ),
                        ],
                      ],
                    ),
                    const SizedBox(height: 8),
                    StatusBadge(
                      text: animal.status.label,
                      color: _getStatusColor(animal.status),
                    ),
                  ],
                ),
              ),
            ],
          ),
        ],
      ),
    );
  }

  Widget _buildInfoSection(Animal animal) {
    final months = DateTime.now().difference(animal.dateOfBirth).inDays ~/ 30;
    final ageStr = months < 12
        ? '$months mo'
        : '${months ~/ 12} yr ${months % 12} mo';

    return AppCard(
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          const Text(
            'Information',
            style: TextStyle(fontSize: 16, fontWeight: FontWeight.bold),
          ),
          const SizedBox(height: 12),
          Wrap(
            spacing: 24,
            runSpacing: 12,
            children: [
              InfoRow(
                label: 'Date of Birth',
                value:
                    '${animal.dateOfBirth.month}/${animal.dateOfBirth.day}/${animal.dateOfBirth.year}',
              ),
              InfoRow(label: 'Age', value: ageStr),
              if (animal.sireTag != null)
                InfoRow(label: 'Sire', value: animal.sireTag!),
              if (animal.damTag != null)
                InfoRow(label: 'Dam', value: animal.damTag!),
              if (animal.purchaseDate != null)
                InfoRow(
                  label: 'Purchase Date',
                  value:
                      '${animal.purchaseDate!.month}/${animal.purchaseDate!.day}/${animal.purchaseDate!.year}',
                ),
              if (animal.purchaseWeight != null)
                InfoRow(
                  label: 'Purchase Weight',
                  value: '${animal.purchaseWeight!.round()} lbs',
                ),
              if (animal.purchasePrice != null)
                InfoRow(
                  label: 'Purchase Price',
                  value: '\$${animal.purchasePrice!.toStringAsFixed(2)}',
                ),
            ],
          ),
        ],
      ),
    );
  }

  Widget _buildCalvingSection(
      BuildContext context, List<CalvingRecord> calvings) {
    return AppCard(
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              const Text(
                'Calving Records',
                style: TextStyle(fontSize: 16, fontWeight: FontWeight.bold),
              ),
              IconButton(
                icon: const Icon(Icons.add_circle, color: AppTheme.earthGreen),
                onPressed: () {
                  // Navigate to add calving record
                },
              ),
            ],
          ),
          if (calvings.isEmpty)
            const Padding(
              padding: EdgeInsets.symmetric(vertical: 16),
              child: Center(
                child: Text(
                  'No calving records',
                  style: TextStyle(color: Colors.grey),
                ),
              ),
            )
          else
            ...calvings.map((record) => _CalvingRecordRow(record: record)),
        ],
      ),
    );
  }

  Widget _buildTreatmentSection(
      BuildContext context, List<TreatmentRecord> treatments) {
    return AppCard(
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              const Text(
                'Treatment Records',
                style: TextStyle(fontSize: 16, fontWeight: FontWeight.bold),
              ),
              IconButton(
                icon: const Icon(Icons.add_circle, color: AppTheme.earthGreen),
                onPressed: () {
                  // Navigate to add treatment record
                },
              ),
            ],
          ),
          if (treatments.isEmpty)
            const Padding(
              padding: EdgeInsets.symmetric(vertical: 16),
              child: Center(
                child: Text(
                  'No treatment records',
                  style: TextStyle(color: Colors.grey),
                ),
              ),
            )
          else
            ...treatments.map((record) => _TreatmentRecordRow(record: record)),
        ],
      ),
    );
  }

  Widget _buildWeightSection(BuildContext context, List<WeightRecord> weights) {
    return AppCard(
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              const Text(
                'Weight Records',
                style: TextStyle(fontSize: 16, fontWeight: FontWeight.bold),
              ),
              IconButton(
                icon: const Icon(Icons.add_circle, color: AppTheme.earthGreen),
                onPressed: () {
                  // Navigate to add weight record
                },
              ),
            ],
          ),
          if (weights.isEmpty)
            const Padding(
              padding: EdgeInsets.symmetric(vertical: 16),
              child: Center(
                child: Text(
                  'No weight records',
                  style: TextStyle(color: Colors.grey),
                ),
              ),
            )
          else
            ...weights.map((record) => _WeightRecordRow(record: record)),
        ],
      ),
    );
  }

  Widget _buildNotesSection(String notes) {
    return AppCard(
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          const Text(
            'Notes',
            style: TextStyle(fontSize: 16, fontWeight: FontWeight.bold),
          ),
          const SizedBox(height: 8),
          Text(notes, style: TextStyle(color: Colors.grey[600])),
        ],
      ),
    );
  }

  Widget _buildDeleteButton(
      BuildContext context, WidgetRef ref, Animal animal) {
    return SizedBox(
      width: double.infinity,
      child: OutlinedButton(
        onPressed: () => _confirmDelete(context, ref, animal),
        style: OutlinedButton.styleFrom(
          foregroundColor: Colors.red,
          side: const BorderSide(color: Colors.red),
        ),
        child: const Text('Delete Animal'),
      ),
    );
  }

  void _confirmDelete(BuildContext context, WidgetRef ref, Animal animal) {
    showDialog(
      context: context,
      builder: (context) => AlertDialog(
        title: const Text('Delete Animal'),
        content: Text(
          'Are you sure you want to delete ${animal.tagNumber}? This action cannot be undone.',
        ),
        actions: [
          TextButton(
            onPressed: () => Navigator.pop(context),
            child: const Text('Cancel'),
          ),
          TextButton(
            onPressed: () async {
              final db = ref.read(databaseProvider);
              await (db.delete(db.animals)
                    ..where((t) => t.id.equals(animal.id)))
                  .go();
              if (context.mounted) {
                Navigator.pop(context);
                context.pop();
              }
            },
            style: TextButton.styleFrom(foregroundColor: Colors.red),
            child: const Text('Delete'),
          ),
        ],
      ),
    );
  }

  IconData _getSexIcon(AnimalSex sex) {
    switch (sex) {
      case AnimalSex.bull:
        return Icons.circle;
      case AnimalSex.cow:
        return Icons.circle_outlined;
      case AnimalSex.heifer:
        return Icons.brightness_1;
      case AnimalSex.steer:
        return Icons.square;
      case AnimalSex.calf:
        return Icons.square_outlined;
    }
  }

  Color _getStatusColor(AnimalStatus status) {
    switch (status) {
      case AnimalStatus.active:
        return Colors.green;
      case AnimalStatus.sold:
        return AppTheme.warmBrown;
      case AnimalStatus.deceased:
      case AnimalStatus.culled:
        return Colors.grey;
      case AnimalStatus.breeding:
        return Colors.blue;
      case AnimalStatus.calving:
        return Colors.purple;
    }
  }
}

class _CalvingRecordRow extends StatelessWidget {
  final CalvingRecord record;

  const _CalvingRecordRow({required this.record});

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 8),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              Text(
                '${record.date.month}/${record.date.day}/${record.date.year}',
                style: const TextStyle(fontWeight: FontWeight.w500),
              ),
              StatusBadge(
                text: 'Ease: ${record.calvingEaseScore}',
                color: _getEaseColor(record.calvingEaseScore),
              ),
            ],
          ),
          const SizedBox(height: 4),
          Row(
            children: [
              const Icon(Icons.tag, size: 14, color: Colors.grey),
              const SizedBox(width: 4),
              Text(record.calfTag, style: const TextStyle(fontSize: 12)),
              const SizedBox(width: 12),
              Text(record.calfSex.label, style: const TextStyle(fontSize: 12)),
              if (record.birthWeight != null) ...[
                const SizedBox(width: 12),
                Text('${record.birthWeight!.round()} lbs',
                    style: const TextStyle(fontSize: 12)),
              ],
            ],
          ),
        ],
      ),
    );
  }

  Color _getEaseColor(int score) {
    switch (score) {
      case 1:
        return Colors.green;
      case 2:
        return Colors.blue;
      case 3:
        return Colors.yellow[700]!;
      case 4:
        return Colors.orange;
      case 5:
        return Colors.red;
      default:
        return Colors.grey;
    }
  }
}

class _TreatmentRecordRow extends StatelessWidget {
  final TreatmentRecord record;

  const _TreatmentRecordRow({required this.record});

  bool get _isWithdrawalActive {
    if (record.withdrawalDate == null) return false;
    return DateTime.now().isBefore(record.withdrawalDate!);
  }

  int get _daysUntilClear {
    if (record.withdrawalDate == null) return 0;
    return record.withdrawalDate!.difference(DateTime.now()).inDays;
  }

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 8),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              Text(
                record.treatment,
                style: const TextStyle(fontWeight: FontWeight.w500),
              ),
              if (_isWithdrawalActive)
                StatusBadge(
                  text: '$_daysUntilClear days',
                  color: Colors.orange,
                ),
            ],
          ),
          const SizedBox(height: 4),
          Row(
            children: [
              Text(
                '${record.date.month}/${record.date.day}/${record.date.year}',
                style: TextStyle(fontSize: 12, color: Colors.grey[600]),
              ),
              const SizedBox(width: 12),
              Text(
                record.dosage,
                style: TextStyle(fontSize: 12, color: Colors.grey[600]),
              ),
              const SizedBox(width: 12),
              Text(
                record.route.label,
                style: TextStyle(fontSize: 12, color: Colors.grey[600]),
              ),
            ],
          ),
          if (record.reason.isNotEmpty) ...[
            const SizedBox(height: 4),
            Text(
              record.reason,
              style: TextStyle(fontSize: 12, color: Colors.grey[600]),
            ),
          ],
        ],
      ),
    );
  }
}

class _WeightRecordRow extends StatelessWidget {
  final WeightRecord record;

  const _WeightRecordRow({required this.record});

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 8),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: [
          Text(
            '${record.date.month}/${record.date.day}/${record.date.year}',
            style: TextStyle(color: Colors.grey[600]),
          ),
          Text(
            '${record.weight.round()} lbs',
            style: const TextStyle(fontWeight: FontWeight.bold),
          ),
        ],
      ),
    );
  }
}

File: swift_reference/Views/Herd/AnimalDetailView.swift
import SwiftUI
import SwiftData

struct AnimalDetailView: View {
    @Environment(\.modelContext) private var modelContext
    @Environment(\.dismiss) private var dismiss

    @Bindable var animal: Animal

    @State private var showingEditAnimal = false
    @State private var showingAddCalving = false
    @State private var showingAddTreatment = false
    @State private var showingDeleteConfirmation = false

    private var dateFormatter: DateFormatter {
        let formatter = DateFormatter()
        formatter.dateStyle = .medium
        return formatter
    }

    var body: some View {
        ScrollView {
            VStack(spacing: Theme.standardPadding) {
                headerSection

                infoSection

                if !animal.calvingRecords.isEmpty || animal.sex == .cow {
                    calvingSection
                }

                treatmentSection

                notesSection

                deleteSection
            }
            .padding(Theme.standardPadding)
        }
        .background(Color(.systemGroupedBackground))
        .navigationTitle("Animal Details")
        .navigationBarTitleDisplayMode(.inline)
        .toolbar {
            ToolbarItem(placement: .navigationBarTrailing) {
                Button("Edit") {
                    showingEditAnimal = true
                }
                .foregroundColor(Theme.earthGreen)
            }
        }
        .sheet(isPresented: $showingEditAnimal) {
            NavigationStack {
                AnimalFormView(animal: animal)
            }
        }
        .sheet(isPresented: $showingAddCalving) {
            NavigationStack {
                CalvingRecordView(record: nil, animal: animal)
            }
        }
        .sheet(isPresented: $showingAddTreatment) {
            NavigationStack {
                TreatmentRecordView(record: nil, animal: animal)
            }
        }
        .confirmationDialog("Delete Animal", isPresented: $showingDeleteConfirmation, titleVisibility: .visible) {
            Button("Delete", role: .destructive) {
                deleteAnimal()
            }
            Button("Cancel", role: .cancel) {}
        } message: {
            Text("Are you sure you want to delete \(animal.tagNumber)? This action cannot be undone.")
        }
    }

    @ViewBuilder
    private var headerSection: some View {
        VStack(spacing: 12) {
            HStack {
                Image(systemName: animal.sex.icon)
                    .font(.system(size: 44))
                    .foregroundColor(Theme.earthGreen)
                    .frame(width: 80, height: 80)
                    .background(Theme.earthGreen.opacity(0.1))
                    .cornerRadius(16)

                VStack(alignment: .leading, spacing: 8) {
                    Text(animal.tagNumber)
                        .font(.title)
                        .fontWeight(.bold)

                    HStack(spacing: 8) {
                        Text(animal.sex.rawValue)
                            .font(.subheadline)
                            .foregroundColor(.secondary)

                        if !animal.breed.isEmpty {
                            Text("•")
                                .foregroundColor(.secondary)
                            Text(animal.breed)
                                .font(.subheadline)
                                .foregroundColor(.secondary)
                        }
                    }

                    Text(animal.status.rawValue)
                        .font(.caption)
                        .fontWeight(.medium)
                        .padding(.horizontal, 10)
                        .padding(.vertical, 4)
                        .background(statusColor.opacity(0.1))
                        .foregroundColor(statusColor)
                        .cornerRadius(6)
                }

                Spacer()
            }

            if animal.treatmentRecords.contains(where: { $0.isWithdrawalActive }) {
                HStack {
                    Image(systemName: "exclamationmark.triangle.fill")
                        .foregroundColor(.orange)
                    Text("Active withdrawal period")
                        .font(.subheadline)
                        .foregroundColor(.orange)
                    Spacer()
                }
                .padding(12)
                .background(Color.orange.opacity(0.1))
                .cornerRadius(8)
            }
        }
        .cardStyle()
    }

    @ViewBuilder
    private var infoSection: some View {
        VStack(alignment: .leading, spacing: 12) {
            Text("Information")
                .font(.headline)

            LazyVGrid(columns: [GridItem(.flexible()), GridItem(.flexible())], spacing: 12) {
                InfoRow(label: "Date of Birth", value: dateFormatter.string(from: animal.dateOfBirth))
                InfoRow(label: "Age", value: animal.ageDescription)

                if let sireTag = animal.sireTag, !sireTag.isEmpty {
                    InfoRow(label: "Sire", value: sireTag)
                }

                if let damTag = animal.damTag, !damTag.isEmpty {
                    InfoRow(label: "Dam", value: damTag)
                }

                if let purchaseDate = animal.purchaseDate {
                    InfoRow(label: "Purchase Date", value: dateFormatter.string(from: purchaseDate))
                }

                if let purchaseWeight = animal.purchaseWeight {
                    InfoRow(label: "Purchase Weight", value: "\(Int(purchaseWeight)) lbs")
                }

                if let purchasePrice = animal.purchasePrice {
                    InfoRow(label: "Purchase Price", value: String(format: "$%.2f", purchasePrice))
                }
            }
        }
        .cardStyle()
    }

    @ViewBuilder
    private var calvingSection: some View {
        VStack(alignment: .leading, spacing: 12) {
            HStack {
                Text("Calving Records")
                    .font(.headline)

                Spacer()

                Button {
                    showingAddCalving = true
                } label: {
                    Image(systemName: "plus.circle.fill")
                        .foregroundColor(Theme.earthGreen)
                }
            }

            if animal.calvingRecords.isEmpty {
                Text("No calving records")
                    .font(.subheadline)
                    .foregroundColor(.secondary)
                    .frame(maxWidth: .infinity)
                    .padding()
            } else {
                ForEach(animal.calvingRecords.sorted { $0.date > $1.date }) { record in
                    CalvingRecordRow(record: record)
                }
            }
        }
        .cardStyle()
    }

    @ViewBuilder
    private var treatmentSection: some View {
        VStack(alignment: .leading, spacing: 12) {
            HStack {
                Text("Treatment Records")
                    .font(.headline)

                Spacer()

                Button {
                    showingAddTreatment = true
                } label: {
                    Image(systemName: "plus.circle.fill")
                        .foregroundColor(Theme.earthGreen)
                }
            }

            if animal.treatmentRecords.isEmpty {
                Text("No treatment records")
                    .font(.subheadline)
                    .foregroundColor(.secondary)
                    .frame(maxWidth: .infinity)
                    .padding()
            } else {
                ForEach(animal.treatmentRecords.sorted { $0.date > $1.date }) { record in
                    TreatmentRecordRow(record: record)
                }
            }
        }
        .cardStyle()
    }

    @ViewBuilder
    private var notesSection: some View {
        if !animal.notes.isEmpty {
            VStack(alignment: .leading, spacing: 12) {
                Text("Notes")
                    .font(.headline)

                Text(animal.notes)
                    .font(.body)
                    .foregroundColor(.secondary)
            }
            .frame(maxWidth: .infinity, alignment: .leading)
            .cardStyle()
        }
    }

    @ViewBuilder
    private var deleteSection: some View {
        Button(role: .destructive) {
            showingDeleteConfirmation = true
        } label: {
            HStack {
                Spacer()
                Text("Delete Animal")
                Spacer()
            }
        }
        .frame(minHeight: Theme.minTapTarget)
        .cardStyle()
    }

    private var statusColor: Color {
        switch animal.status {
        case .active: return .green
        case .sold: return Theme.warmBrown
        case .deceased: return .gray
        case .culled: return .orange
        case .breeding: return .blue
        case .calving: return .purple
        }
    }

    private func deleteAnimal() {
        modelContext.delete(animal)
        dismiss()
    }
}

struct InfoRow: View {
    let label: String
    let value: String

    var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            Text(label)
                .font(.caption)
                .foregroundColor(.secondary)

            Text(value)
                .font(.subheadline)
                .fontWeight(.medium)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }
}

struct CalvingRecordRow: View {
    let record: CalvingRecord

    private var dateFormatter: DateFormatter {
        let formatter = DateFormatter()
        formatter.dateStyle = .medium
        return formatter
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack {
                Text(dateFormatter.string(from: record.date))
                    .font(.subheadline)
                    .fontWeight(.medium)

                Spacer()

                Text("Ease: \(record.calvingEaseScore)")
                    .font(.caption)
                    .padding(.horizontal, 8)
                    .padding(.vertical, 2)
                    .background(calvingEaseColor.opacity(0.1))
                    .foregroundColor(calvingEaseColor)
                    .cornerRadius(4)
            }

            HStack(spacing: 16) {
                HStack(spacing: 4) {
                    Image(systemName: "tag.fill")
                        .font(.caption)
                        .foregroundColor(.secondary)
                    Text(record.calfTag)
                        .font(.caption)
                }

                HStack(spacing: 4) {
                    Image(systemName: record.calfSex.icon)
                        .font(.caption)
                        .foregroundColor(.secondary)
                    Text(record.calfSex.rawValue)
                        .font(.caption)
                }

                if let weight = record.birthWeight {
                    HStack(spacing: 4) {
                        Image(systemName: "scalemass.fill")
                            .font(.caption)
                            .foregroundColor(.secondary)
                        Text("\(Int(weight)) lbs")
                            .font(.caption)
                    }
                }
            }
            .foregroundColor(.secondary)
        }
        .padding(.vertical, 8)
    }

    private var calvingEaseColor: Color {
        switch record.calvingEaseScore {
        case 1: return .green
        case 2: return .blue
        case 3: return .yellow
        case 4: return .orange
        case 5: return .red
        default: return .gray
        }
    }
}

struct TreatmentRecordRow: View {
    let record: TreatmentRecord

    private var dateFormatter: DateFormatter {
        let formatter = DateFormatter()
        formatter.dateStyle = .medium
        return formatter
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack {
                Text(record.treatment)
                    .font(.subheadline)
                    .fontWeight(.medium)

                Spacer()

                if record.isWithdrawalActive {
                    Text("\(record.daysUntilClear) days")
                        .font(.caption)
                        .padding(.horizontal, 8)
                        .padding(.vertical, 2)
                        .background(Color.orange.opacity(0.1))
                        .foregroundColor(.orange)
                        .cornerRadius(4)
                }
            }

            HStack(spacing: 16) {
                Text(dateFormatter.string(from: record.date))
                    .font(.caption)
                    .foregroundColor(.secondary)

                Text(record.dosage)
                    .font(.caption)
                    .foregroundColor(.secondary)

                Text(record.route.rawValue)
                    .font(.caption)
                    .foregroundColor(.secondary)
            }

            if !record.reason.isEmpty {
                Text(record.reason)
                    .font(.caption)
                    .foregroundColor(.secondary)
            }
        }
        .padding(.vertical, 8)
    }
}

#Preview {
    NavigationStack {
        AnimalDetailView(animal: Animal(tagNumber: "A001", breed: "Angus", sex: .cow, status: .active))
    }
    .modelContainer(for: [Animal.self, CalvingRecord.self, TreatmentRecord.self], inMemory: true)
}
